ui/src/ScholarReader.tsx

```diff
@@ -182,6 +196,20 @@ class ScholarReader extends React.PureComponent<ScholarReaderProps, State> {
     }
   }
 
+  async addToLibrary(paperId: string, paperTitle: string) {
+    if (this.props.paperId !== undefined) {
```

Comment:
could just do if (this.props.paperId) and it'll understand if its not null/undefined